tasks/windows.ps1

```diff
@@ -0,0 +1,195 @@
+[CmdletBinding()]
+Param(
+  [Parameter(Mandatory = $True)]
+  [String]
+  $Master,
+
+  [Parameter(Mandatory = $False)]
+  [ValidateScript({ ![String]::IsNullOrEmpty($_) })]
+  [String]
+  $CACert_Content,
+
+  [Parameter(Mandatory = $False)]
+  [ValidateScript({ if ([System.URI]::CheckHostName($_) -eq 'Unknown') {
+      throw "Invalid DNS name `"$_`""
+  }; $True})]
+  [String]
+  $CertName,
+
+  [Parameter(Mandatory = $False)]
+  [ValidateScript({ $_ -split ',' | % {
+    if ([System.URI]::CheckHostName($_) -eq 'Unknown') {
+      throw "Invalid DNS name `"$_`""
+  }}; $True})]
+  [String]
+  $DNS_Alt_Names
+)
+
+function Get-HostName
+{
+  $ipAddress = ([System.Net.Dns]::GetHostEntry([System.Environment]::MachineName)).AddressList |
```

Comment:
This will result in more than one ipAddress on a system with multiple nics/addresses. You will have to filter further to find the one with the default route and is not disconnected. `Get-NetIPConfiguration` gives you that information easily, but IIRC is only available on 2012 and above, but not sure.

Reply:
Ah yeah, good catch - thanks... Haven't tested on a system with multiple NICs yet, but this would have crashed and burned.

We might have to stick with a `-First 1` on the `Select` for now and hope for the best. Ideally, the end user supplies the value to the task anyhow (the DIG case) rather than us trying to disambiguate multiple possibilities (the best thing may be to use the fallback if there's only one, and fail the task if more?). This also almost certainly doesn't work as implemented for AWS / Azure / GCE. It's pretty naive, but at least gives a correct hostname in VMPooler and Vagrant.

I'm having deja vu - https://github.com/puppetlabs/facter/pull/537